.github/workflows/endor_scan_and_generate_sbom.yml

```
@@ -0,0 +1,144 @@
+name: Endor Labs Scan and Generate SBOM
+
+# This workflow runs an SCA scan using Endor Labs.
+# When triggered by a pull_request
+#   a PR scan is run (non-blocking) that reports any identified issues as a PR comment.
+# When triggered by a create, push or workflow_dispatch
+#   a monitoring scan is run, an SBOM file is generated, and if the SBOM has changed a PR is opened.
+# The pull_request and push triggers are scoped to cmake files in the master and release branches
+# Internal documentation: go/sbom-scope
+
+on:
+  workflow_dispatch: {}
+  pull_request:
+    branches:
+      - "master"
+      - "releases/v*"
+      - "debian/*"
+    paths:
+      - "**/CMakeLists.txt"
+      - "**/*.cmake"
+  push:
+    branches:
+      - "master"
+      - "releases/v*"
+      - "debian/*"
+    paths:
+      - "**/CMakeLists.txt"
+      - "**/*.cmake"
+
+permissions:
+  id-token: write # Required to request a json web token (JWT) for keyless authentication with Endor Labs
+  contents: write # Required for commit
+  pull-requests: write # Required for PR
+
+jobs:
+  endor_scan_and_generate_sbom:
+    runs-on: ubuntu-latest
+
+    concurrency:
+      group: sbom-${{ github.ref }}
+      cancel-in-progress: false
+
+    env:
+      # for a pull_request, we run a PR scan and do not generate an SBOM
+      PR_SCAN: ${{ github.event_name == 'pull_request' }}
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v6
+        with:
+          fetch-tags: true
+          submodules: recursive
+
+      - name: Configure CMake and fetch dependency sources
+        env:
+          BUILD_TYPE: Release
+          BUILD: ${{github.workspace}}/build
+          CXX_STANDARD: 17
+        working-directory: ${{env.BUILD}}
+        run: |
+          cmake .. -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} -DCMAKE_CXX_STANDARD=${{env.CXX_STANDARD}} -DENABLE_TESTS=ON
+          rm .gitignore # prevent exclusion of build/_deps from endorctl scan
```

```suggestion
          echo "!_deps" >>.gitignore # prevent exclusion of build/_deps from endorctl scan
```
Consider excluding the `_deps` directory specifically instead.